File: UIStepperObjC/UIStepperObjC/ViewController.m
//
//  ViewController.m
//  UIStepperObjC
//
//  Created by Field Employee on 3/26/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

UILabel *label;
UIStepper *stepper;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)updateView:(UIStepper*)stepperOne {
    label.text = [NSString stringWithFormat:@"Counter: %d", (int)stepper.value];
}

- (void)loadView {
    UIView *view = [[UIView alloc] init];
    view.backgroundColor = UIColor.whiteColor;
    
    label = [[UILabel alloc] init];
    
    stepper = [[UIStepper alloc] init];
    stepper.tintColor = UIColor.whiteColor;
    [stepper addTarget:self action:@selector(updateView:) forControlEvents:UIControlEventValueChanged];
    stepper.value = 15;
    stepper.minimumValue = -100;
    stepper.maximumValue = 100;
    stepper.stepValue = 5;
    
    [view addSubview:label];
    [view addSubview:stepper];
    
    self.view = view;
    [self updateView:nil];
    
    // Layout
    label.translatesAutoresizingMaskIntoConstraints = false;
    stepper.translatesAutoresizingMaskIntoConstraints = false;
    
    [NSLayoutConstraint activateConstraints:@[
        [label.topAnchor constraintEqualToAnchor:view.topAnchor constant:40.0],
        [label.leadingAnchor constraintEqualToAnchor:view.leadingAnchor constant:40.0],

        [stepper.leadingAnchor constraintEqualToAnchor:label.trailingAnchor constant:20.0],
        [stepper.centerYAnchor constraintEqualToAnchor:label.centerYAnchor]
     ]
    ];
}


@end

File: UIStepperSwift/UIStepperSwift/ViewController.swift
//
//  ViewController.swift
//  UIStepperSwift
//
//  Created by Field Employee on 3/26/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var label : UILabel!
    var stepper : UIStepper!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    override func loadView() {
        // UI
        let view = UIView()
        view.backgroundColor = .red
        
        label = UILabel()

        stepper = UIStepper()
        stepper.tintColor = .white
        stepper.addTarget(self, action: #selector(updateView), for: .valueChanged)
        stepper.value = 15
        stepper.minimumValue = -100
        stepper.maximumValue = 100
        stepper.stepValue = 5

        view.addSubview(label)
        view.addSubview(stepper)
        
        self.view = view
        self.updateView()
        
        // Layout
        label.translatesAutoresizingMaskIntoConstraints = false
        stepper.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([
            label.topAnchor.constraint(equalTo: view.topAnchor, constant: 40),
            label.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 40),

            stepper.leadingAnchor.constraint(equalTo: label.trailingAnchor, constant: 20),
            stepper.centerYAnchor.constraint(equalTo: label.centerYAnchor),
        ])
        
    }
    
    @objc func updateView() {
        label.text = "Counter: \(Int(stepper.value))"
    }
    
}

